core/sr-io/without_std.rs

```diff
@@ -132,9 +132,7 @@ pub fn storage(key: &[u8]) -> Option<Vec<u8>> {
 		if length == u32::max_value() {
 			None
 		} else {
-			let ret = slice::from_raw_parts(ptr, length as usize).to_vec();
-			ext_free(ptr);
-			Some(ret)
+			Some(<Vec<u8>>::from_raw_parts(ptr, length as usize, length as usize))
```

Comment:
As was discussed in the issue, this actually violates formal preconditions required by `Vec::from_raw_parts` (i.e. the pointer is allocated not with `Vec` routines). Given that we agree with it, I still would like to have a comment here mentioning this fact.

And there should be another case nearby.

Reply:
I'm still against using `Vec::from_raw_parts` because of all the invariants. If we can prove the invariants, I'm happy.

Reply:
@gavofyork What is your opinion on this?

Reply:
It may be better to return some other new type as well.

Reply:
As I stated in the other thread I'm against this UB dogma. I demonstrated that this code is safe. If you think otherwise please bring the argument forward, otherwise this code goes in.

Reply:
Yes, I'd agree that it would work for us and I don't have anymore to add to our discussion here! But I'd still be in favor of a comment here. I hope it will save us a headache.

Also, @cmichi there is another case where we apply the same pattern: `child_storage`, a function below. I'd like them to be synchronized.